ssz/src/main/java/tech/pegasys/teku/ssz/backing/tree/TreeUtil.java

```diff
@@ -77,7 +77,7 @@ public static TreeNode createDefaultTree(long maxLength, TreeNode defaultNode) {
   }
 
   /** Creates a binary tree of width `nextPowerOf2(leafNodes.size())` with specific leaf nodes */
-  public static TreeNode createTree(List<TreeNode> leafNodes) {
+  public static TreeNode createTree(List<? extends TreeNode> leafNodes) {
```

Comment:
nit - these nodes aren't necessarily terminal leaf nodes:
```suggestion
  public static TreeNode createTree(List<? extends TreeNode> children) {
```

Reply:
Yep, still trying to find the correct term for such nodes, but right `children` is closer than `leaf` here